tests/0150-telemetry_mock.c

```diff
@@ -197,6 +197,74 @@ void do_test_telemetry_get_subscription_push_telemetry(void) {
         SUB_TEST_PASS();
 }
 
+
+/**
+ * @brief Tests the 'happy path' of GetTelemetrySubscriptions, followed by
+ *        successful PushTelemetry requests.
+ *        See `requests_expected` for detailed expected flow.
+ */
+void do_test_telemetry_get_subscription_push_telemetry_consumer(void) {
+        rd_kafka_conf_t *conf;
+        const char *bootstraps;
+        rd_kafka_mock_cluster_t *mcluster;
+        char *expected_metrics[]           = {"*"};
+        rd_kafka_t *consumer               = NULL;
+        rd_kafka_mock_request_t **requests = NULL;
+        size_t request_cnt;
+        const int64_t push_interval = 5000;
+
+        rd_kafka_telemetry_expected_request_t requests_expected[] = {
+            /* T= 0 : The initial GetTelemetrySubscriptions request. */
+            {.ApiKey           = RD_KAFKAP_GetTelemetrySubscriptions,
+             .broker_id        = -1,
+             .expected_diff_ms = -1,
+             .jitter_percent   = 0},
+            /* T = push_interval + jitter : The first PushTelemetry request */
+            {.ApiKey           = RD_KAFKAP_PushTelemetry,
+             .broker_id        = -1,
+             .expected_diff_ms = push_interval,
+             .jitter_percent   = 20},
+            /* T = push_interval*2 + jitter : The second PushTelemetry request.
+             */
+            {.ApiKey           = RD_KAFKAP_PushTelemetry,
+             .broker_id        = -1,
+             .expected_diff_ms = push_interval,
+             .jitter_percent   = 0},
+        };
+
+        SUB_TEST();
+
+        mcluster = test_mock_cluster_new(1, &bootstraps);
+        rd_kafka_mock_telemetry_set_requested_metrics(mcluster,
+                                                      expected_metrics, 1);
+        rd_kafka_mock_telemetry_set_push_interval(mcluster, push_interval);
+        rd_kafka_mock_start_request_tracking(mcluster);
+
+        test_conf_init(&conf, NULL, 30);
+        test_conf_set(conf, "bootstrap.servers", bootstraps);
+        test_conf_set(conf, "debug", "telemetry");
+        consumer = test_create_handle(RD_KAFKA_CONSUMER, conf);
+
+        /* Poll for enough time for two pushes to be triggered, and a little
+         * extra, so 2.5 x push interval. */
+        test_poll_timeout(consumer, push_interval * 2.5);
```

Comment:
`test_poll_timeout` should call `test_consumer_poll_timeout` with the consumer or `test_produce_msgs` with the consumer so we can see metric in logs that are not zero and check if they're correct. A later refactor would be to automatically check the metric values but for this step that is enough.